azure/README.md

```diff
@@ -0,0 +1,197 @@
+
+### (Optional) Multi-user setup: shared state
+
+If you want to host a multi-user setup, you will probably want to share the state file so that everyone can try related challenges. We have provided a starter to easily do so using an Azure storage container.
+
+First, enable the `Microsoft.Storage` API (if it isn't already) using:
+
+```bash
+az provider register --namespace Microsoft.Storage
+```
+
+Then, apply the Terraform (optionally add `-var="region=YOUR_DESIRED_REGION"` to the apply to use a region other than the default `East US`):
+
+```bash
+cd shared-state
+terraform init
+terraform apply
+```
+
+The storage account name should be in the output. Please use that to configure the Terraform backend in `main.tf` by uncommenting the part on the `backend "azurerm"`.
+
+**Note**: You'll need to follow the description below for the "existing resource group" i.e., use the `data.azurerm_resource_group.default` resource.
+
+
+### WrongSecrets-ctf-party
+
+1. Set either a new resource group or use an existing resource group in `main.tf` (it defaults to the existing `OWASP-Projects` resource group). Note that you'll need to find/replace references to "data.azurerm_resource_group.default" to "arurerm_resource_group.default" if you want to create a new one.
+2. check whether you have the right project by doing `az account show` (after `az login`). Want to set the project as your default? Use `az account set --subscription <.id here>`.
+3. If not yet enabled, register the required services for the subscription, run:
+    - `az provider register --namespace Microsoft.ContainerService`
+    - `az provider register --namespace Microsoft.KeyVault`
+    - `az provider register --namespace Microsoft.ManagedIdentity`
+4. Run `terraform init` (if required, use `tfenv` to select TF 0.14.0 or higher )
+5. Run `terraform plan` to see what will be created (optional).
+6. Run `terraform apply`. Note: the apply will take 5 to 20 minutes depending on the speed of the Azure backplane.
+7. Run `./build-and-deploy-azure.sh`. Your kubeconfig file will automatically be updated.
+
+Your AKS cluster should be visible in your resource group. Want a different region? You can modify `terraform.tfvars` or input it directly using the `region` variable in plan/apply.
+
+Are you done playing? Please run `terraform destroy` twice to clean up.
+
+### Test it
+
+When you have completed the installation steps, you can do `kubectl port-forward service/wrongsecrets-balancer 3000:3000` and then go to [http://localhost:3000](http://localhost:3000).
+
+Want to know how well your cluster is holding up? Check with
+
+```sh
+    kubectl top nodes
+    kubectl top pods
+```
+
```

Comment:
should we say something here about how we can influence the main image shown on top? and what texts to show at the balancer ui?

Reply:
Done for both Azure and AWS readme.